nhost/service/manager.go

```diff
@@ -86,13 +95,162 @@ func (m *dockerComposeManager) SetGitBranch(gitBranch string) {
 	m.composeConfig = compose.NewConfig(m.nhostConfig, m.ports, m.env, gitBranch, m.projectName)
 }
 
+func (m *dockerComposeManager) checkPorts(ctx context.Context) error {
```

Comment:
This method is too long and its intention is very hard to read. As a rule of thumb methods should fit on the screen. (60 lines is usually a good measure) and be comprised of intentions rather than a collection of steps to get there. For instance:

```
1. getPortsForExistingServices(...)
2. getRandomPort(...)
3. isPortAvailable(...)
```

Getting only into the "low-level" instructions as late as possible (preferably when they only serve one purpose, like checking if a given port is available)